src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -208,6 +210,21 @@ async static (state) =>
 
                     FindLibraryEntryResult refItemResult = await refItemTask;
 
+                    if (importRefItem.LibraryRangeIndex == rootProjectRefItem.LibraryRangeIndex)
```

Comment:
For my own learning, what does the LibraryRangeIndex contain?

Reply:
This is part of the intering that the new resolver uses.  Each VersionRange gets assigned an integer, for example:

| VersionRange | LibraryRangeIndex |
|---|---|
| Project1 1.0.0 | 0 |
| PackageA 1.0.0 | 1 |
| PackageB 1.0.0 | 2 |
| PackageC 1.0.0 | 3 |
| PackageC 2.0.0 | 4 |
| PackageC 3.0.0 | 5 |

This is used in dictionaries so that lookups are quick, a version range is transformed into an integer and then it is looked up.